packages/eslint-config/test/__snapshots__/typescript-eslint.spec.js.snap

```diff
@@ -157,6 +166,9 @@ Object {
     "@typescript-eslint/no-implied-eval": Array [
       "error",
     ],
+    "@typescript-eslint/no-import-type-side-effects": Array [
+      "error",
+    ],
```

Comment:
New rule: https://typescript-eslint.io/rules/no-import-type-side-effects

Thought this would actually be favorable in adding in. Let me know your thoughts.